src/bcs/serializable/move-structs.ts

```diff
@@ -66,8 +88,21 @@ export class MoveVector<T extends Serializable> extends Serializable {
    * @params values: an array of `numbers` to convert to U8s
    * @returns a `MoveVector<U8>`
    */
-  static U8(values: Array<number>): MoveVector<U8> {
-    return new MoveVector<U8>(values.map((v) => new U8(v)));
+  static U8(values: Array<number> | HexInput): MoveVector<U8> {
+    let numbers: Array<number>;
+
+    if (Array.isArray(values) && typeof values[0] === "number") {
```

Comment:
we assume all elements in the array are numbers and not just the first one? or we actually check it somewhere?

Reply:
each one is eventually validated when you map the numbers I believe, but will check it here if not

Reply:
it's validated when mapping the numbers, since `new U8(v)` should validate it., but looks like it might actually inadvertently accept an array of strings due to the way bigint works in the number validation.

Ah yes, I already have a test case for it in `bcs-helper.test.ts`:

```typescript
      expect(() => MoveVector.U8(["01", "02", "03"] as any)).toThrow();
```